library/block/block_content_registry/src/main/java/org/quiltmc/qsl/block/content/registry/impl/BlockContentRegistriesClientInitializer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.block.content.registry.impl;
+
+import org.quiltmc.loader.api.ModContainer;
+import org.quiltmc.loader.api.QuiltLoader;
+import org.quiltmc.qsl.base.api.entrypoint.client.ClientModInitializer;
+import org.quiltmc.qsl.block.content.registry.api.BlockContentRegistries;
+import org.quiltmc.qsl.tooltip.api.client.ItemTooltipCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.text.Text;
+
+
+public class BlockContentRegistriesClientInitializer implements ClientModInitializer {
+	public static final Logger LOGGER = LoggerFactory.getLogger("BlockContentRegistriesClientInitializer");
+
+	public static final String ENABLE_TOOLTIP_DEBUG = "quilt.block.block_content_registry.enable_tooltip_debug";
+
+	@Override
+	public void onInitializeClient(ModContainer mod) {
+		if (Boolean.getBoolean(ENABLE_TOOLTIP_DEBUG) && QuiltLoader.isModLoaded("quilt_tooltip")) {
+			ItemTooltipCallback.EVENT.register((stack, player, context, lines) -> {
+				Item item = stack.getItem();
+
+				Block block = Block.getBlockFromItem(item);
```

Comment:
Nitpick: I'd group this together with the item var declaration, although thinking about it, since you don't use it elsewhere, perhaps you could inline it?